src/data-controller.ts

```diff
@@ -11,14 +11,18 @@ function generateProductHTML(product: Product): string {
 
 function renderProducts(prods: Product[]): void {
     const container = document.getElementById("main-container");
+    if (container) container.innerHTML = "";
```

Comment:
This is repetitive with line 16. See below comment

Reply:
Fixed.